### PR TITLE
Pmail - HTML and CSS

### PR DESCRIPTION
**Pull request by paulh16**
This pull request is for the HTML and CSS components for the Pmail exercise. I am going to do the JavaScript component separately because I want to learn it well and spend some time on it.

For this project, I used Grunt as my build tool. I also used icon fonts and SVGs. However, they will not display on the webpage unless it is running on a local server. Here is a link to Github Pages where I've placed the site:

http://paulh16.github.io/level_up_exercises/

I used HTML5 Boilerplate as my base HTML, CSS and JS, and SASS as my CSS pre-processor. The structure of my SASS files is as follows.

The `styles.scss` consists of the HTML5 Boilerplate styles plus mine, which I import on line 115. I also import `normalize.css` on line 17. The `imports.scss` file is responsible for importing my partial SASS files. In accordance with SMACCS, I broke out the files into give groups: base; components; helpers; layout; and state. Each folder has multiple SASS partials that are named according to their purpose.

Thank you for reviewing. Please let me know if you have any questions.

### CODE REVIEW

File: pmail/css/build/main_unprefixed.css.map [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (danvayn) [inline pmail/css/build/main_unprefixed.css.map]: i probably would've put the started js work in another branch...

File: pmail/humans.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# humanstxt.org/
```
Comment (danvayn) [inline pmail/humans.txt]: its important to delete generated files if they really serve no purpose to you or your code.

File: pmail/index.html [hunk 3]
```diff
@@ -0,0 +1,418 @@
+<!doctype html>
+<html class="no-js" lang="">
```
Comment (danvayn) [inline pmail/index.html]: define yo lang!

File: pmail/index.html [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+    <head>
+        <meta charset="utf-8">
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <title>Pmail</title>
+        <meta name="description" content="This is the new Pmail web application for email!">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+
```
Comment (danvayn) [inline pmail/index.html]: an important thing to keep in mind when working with html/css is that blank lines should have significance. use them to distinguish different blocks of code, but only if necessary. the rest of the file tends to have a lot of random blank lines, i'd manage those if i were you.

File: pmail/index.html [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+            <div class="login">
+
+                <button class="grid-button control-button">
+                    <svg class="icon">
+                      <use xlink:href="img/sprite.svg#grid-three-up" class="icon-grid-three-up"></use>
+                    </svg>
+                </button>
+
+                <p>phaddad@enova.com
+                    <svg class="icon">
```
Comment (danvayn) [inline pmail/index.html]: id wrap these within another div to make it easier to interact with later

File: pmail/index.html [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+                <div class="app-control">
+                    <button id="mail-dropdown" class="button no-background">Mail
+                        <svg class="icon">
+                          <use xlink:href="img/sprite.svg#caret-bottom" class="icon-caret-bottom"></use>
+                        </svg>
+                    </button>
+                </div>
+
+                <div class="mailboxes">
+                    <button class="red-button">Compose</button>
```
Comment (danvayn) [inline pmail/index.html]: red-button isn't a very good class name :)

File: pmail/index.html [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+
+                    <ul>
+                        <a href="#"><li class="current">Inbox</li></a>
```
Comment (danvayn) [inline pmail/index.html]: incorrect order of tags

File: pmail/index.html [hunk 8]
```diff
@@ -0,0 +1,418 @@
...
+                    </ul>
+                </div>
+
+            </div>
+
+
+            <div class="email-container col-flex">
+
+                <div class="email-controls clearfix">
+                    <div>
```
Comment (danvayn) [inline pmail/index.html]: empty divs are never the solution! you could either create a second email controls div or give the buttons a class that handles your floats for you.

File: pmail/index.html [hunk 9]
```diff
@@ -0,0 +1,418 @@
...
+                        <button class="more-button control-button">More
+                            <svg class="icon">
+                              <use xlink:href="img/sprite.svg#caret-bottom" class="icon-caret-bottom"></use>
+                            </svg>
+                        </button>
+
+                    </div>
+
+                    <div>
+                        <button class="pagination-button control-button">1&mdash;50 of 396</button>
```
Comment (danvayn) [inline pmail/index.html]: making this a button adds a couple little problems to your functionality (i don't believe it's supposed to have a border on hover for example) -- tend to it well.

File: pmail/index.html [hunk 10]
```diff
@@ -0,0 +1,418 @@
...
+                            </svg>
+                        </button>
+
+                    </div>
+                </div>
+
+                <section class="emails">
+                    <table>
+
+                        <tr>
```
Comment (danvayn) [inline pmail/index.html]: if you use tables for your emails, make sure you're keeping tbody and colgroup around just to be proper.

File: pmail/index.html [hunk 11]
```diff
@@ -0,0 +1,418 @@
...
+                            <td>
+                                <span class="fa fa-square-o fa-lg"></span>
```
Comment (danvayn) [inline pmail/index.html]: pass a class to your tds housing similar objects in each line just to make targeting them easier, and so others can understand their purpose. also, font-awesome says that modifying the parent's font-size will change the icons size, so you can always keep that in mind.

File: pmail/index.html [hunk 12]
```diff
@@ -0,0 +1,418 @@
...
+                            </td>
+                            <td>
+                                <span class="fa fa-star-o fa-lg"></span>
+                            </td>
+                            <td>Marks, Jonathan</td>
```
Comment (danvayn) [inline pmail/index.html]: same with these for the above comment, if you ever wanted to target them you'd have to do something like "look for fourth td" instead of better solutions

File: pmail/index.html [hunk 13]
```diff
@@ -0,0 +1,418 @@
...
+            </div>
+
+        </div>
+
+        <script src="//ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
+        <script>window.jQuery || document.write('<script src="js/vendor/jquery-1.11.2.min.js"><\/script>')</script>
+        <script src="js/main.js"></script>
+
+        <!-- Google Analytics: change UA-XXXXX-X to be your site's ID. -->
+        <script>
```
Comment (danvayn) [inline pmail/index.html]: if you aren't using something like google analytics the better solution is to delete stuff like this. it goes back to using scaffolding setups like these -- when you use em, you should fully understand what you're generating and how to strip it down as soon as you can.

File: pmail/js/build/production.js [hunk 14]
```diff
@@ -0,0 +1,26 @@
+
+
+// Avoid `console` errors in browsers that lack a console.
+(function() {
```
Comment (danvayn) [inline pmail/js/build/production.js]: hmm..

File: pmail/Gruntfile.js [hunk 15]
```diff
@@ -0,0 +1,110 @@
+module.exports = function(grunt) {
+
+  // Project configuration.
+  grunt.initConfig({
+    pkg: grunt.file.readJSON('package.json'),
```
Comment (danvayn) [inline pmail/Gruntfile.js]: when using grunt make sure you separate production and developer modes! (http://davidtucker.net/articles/automating-with-grunt/)

File: pmail/sass/_normalize.scss [hunk 16]
```diff
@@ -0,0 +1,431 @@
+/*------------------------------------*\
```
Comment (danvayn) [inline pmail/sass/_normalize.scss]: id separate files like these to their own sass subfolder since you'll never want to look at them really

File: pmail/sass/components/_buttons.scss [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+  @include font(header);
+  @include font-size(1.15);
+  color: $orange-red;
+  font-weight: 400;
+  background: none;
+  padding: .3rem;
+}
+
+.red-button {
+  @extend .button;
```
Comment (danvayn) [inline pmail/sass/components/_buttons.scss]: you could just like pass the button class to the actual red button, right?

File: pmail/sass/components/_buttons.scss [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+  background: $light-gray-gradient;
+  //border: 1px solid lighten($dark-gray, .8);
+  @include border(lighten($dark-gray, .8));
+  border-radius: 2px;
+  padding: .4rem;
+  margin-right: 0.8rem;
+  @include font-size(0.7);
+
+  &:last-of-type {
```
Comment (danvayn) [inline pmail/sass/components/_buttons.scss]: if im not mistaken targeting the child of the parent would be a better approach to eliminating margin-right, rather than keeping it all within the child.

File: pmail/sass/components/_buttons.scss [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+  @include font-size(.7);
+  line-height: 1.4;
+}
+
+.pagination-button {
+  background: none;
+  border: none;
+  margin-right: 0;
+  @include font(body-bold);
+  @include font-size(.75);
```
Comment (danvayn) [inline pmail/sass/components/_buttons.scss]: i think its alright to plug a lot of includes within your classes, but make sure you keep your rules in the same order as the way you did it with similar selectors

File: pmail/sass/components/_buttons.scss [hunk 20]
```diff
@@ -0,0 +1,98 @@
+/*------------------------------------*\
+    #COMPONENTS - BUTTONS
```
Comment (danvayn) [inline pmail/sass/components/_buttons.scss]: so i think the better approach is to use your buttons partial within components to be intended as a component -- like writing mixins and whatnot. that's part of the fun of writing SASS -- your mixins can include other mixins :xzibit:

File: pmail/sass/components/_email_controls.scss [hunk 21]
```diff
@@ -0,0 +1,18 @@
+/*------------------------------------*\
+    #COMPONENTS - EMAIL CONTROLS
+\*------------------------------------*/
+
+.email-controls {
+  padding: .5rem 0;
+  border-bottom: 1px $dark-gray solid;
+
+  div {
```
Comment (danvayn) [inline pmail/sass/components/_email_controls.scss]: *cringes*

File: pmail/sass/components/_email_table.scss [hunk 22]
```diff
@@ -0,0 +1,48 @@
+/*------------------------------------*\
+    #COMPONENTS - EMAIL TABLE
+\*------------------------------------*/
+
+.emails {
+
+  background: $light-gray-gradient;
+
+  table {
```
Comment (danvayn) [inline pmail/sass/components/_email_table.scss]: passing a class here would make appending tr's to table more sensible if you ever did that in the future

File: pmail/sass/components/_icons.scss [hunk 23]
```diff
@@ -0,0 +1,132 @@
...
+
+  use {
+    fill: $button-gray;
+  }
+}
+
+.selection-button {
+
+  svg:first-of-type {
+    @include icon-size(14px, 14px);
```
Comment (danvayn) [inline pmail/sass/components/_icons.scss]: as i said earlier i think you can control icon-sizes from the parent, making your job a lot easier here

File: pmail/sass/components/_icons.scss [hunk 24]
```diff
@@ -0,0 +1,132 @@
...
+      @include icon-size(8px, 8px);
+    }
+  }
+}
+
+.fa-star {
+  color: $star-yellow;
+}
+
+.fa-star-o, .fa-square-o {
```
Comment (danvayn) [inline pmail/sass/components/_icons.scss]: do you have to use these class names?

File: pmail/sass/components/_navigation.scss [hunk 25]
```diff
@@ -0,0 +1,32 @@
+/*------------------------------------*\
+    #COMPONENTS - NAVIGATION
+\*------------------------------------*/
+
```
Comment (danvayn) [inline pmail/sass/components/_navigation.scss]: i didnt say it earlier but i do like how you laid out your css, great job

File: pmail/index.html [hunk 26]
```diff
@@ -0,0 +1,418 @@
...
+
+                    <div class="col-1-4 capacity">
+                        <p>0% full</p>
+                        <p>Using 0.17 GB of your 30 GB</p>
+                    </div>
+
+                    <nav class="col-1-2 secondary-links">
+
+                        <nav>
+                            <ul>
```
Comment (danvayn) [inline pmail/index.html]: aaaaaaaaaah you have a nav within a nav

File: pmail/sass/layout/_mediaqueries.scss [hunk 27]
```diff
@@ -0,0 +1,28 @@
+/*------------------------------------*\
```
Comment (danvayn) [inline pmail/sass/layout/_mediaqueries.scss]: consistent tiles yo, _media-queries.scss!

File: pmail/sass/layout/_mediaqueries.scss [hunk 28]
```diff
@@ -0,0 +1,28 @@
...
+    #LAYOUT - MEDIA QUERIES
+\*------------------------------------*/
+
+@mixin bp($point) {
+
+  $bp-alpha: "(min-width: 5rem)";
```
Comment (danvayn) [inline pmail/sass/layout/_mediaqueries.scss]: i know mr chris coyier said you could name these anything but numbers but really these aren't very indicative names (like smallest-size > alpha-size > bp-alpha)

File: pmail/sass/layout/_sidebar.scss [hunk 29]
```diff
@@ -0,0 +1,40 @@
...
+    }
+  }
+
+  .mailboxes {
+    padding: .5rem 0;
+    @include font-size(0.85);
+    @include font(body);
+
+    button {
+      margin-left: 0.5rem;
```
Comment (danvayn) [inline pmail/sass/layout/_sidebar.scss]: careful about all these rems that house ems within them

File: pmail/sass/state/_links.scss [hunk 30]
```diff
@@ -0,0 +1,21 @@
+.sidebar-container {
+
+  .current {
+    @include font(body-bold);
+    color: #d14836;
+
+    &::before {
```
Comment (danvayn) [inline pmail/sass/state/_links.scss]: just use one colon (http://www.impressivewebs.com/before-after-css3/)

File: pmail/sass/styles.scss [hunk 31]
```diff
@@ -0,0 +1,286 @@
+/*------------------------------------*\
```
Comment (danvayn) [inline pmail/sass/styles.scss]: gross, this guy's just sitting around

**Comment by danvayn** [comment]
heyyy im done, great work dude
